### DISCUSSION TITLE
How many predictors can I include in ANCOM-BC?

### DISCUSSION THREAD

**Discussion by simone-anza** [FrederickHuangLin/ANCOMBC#60] [Q&A] [answered]
Hi,

thank you for developing such a great tool! I am wondering whether there is an "optimal" number of predictors or a "**limit**" in the number of predictors we can include in ANCOM-BC according to sample size. I am thinking about something similar to GLMM and the suggestion to have at least 7-10 data points per predictor (e.g. for a dataset of 40data points it's not recommended to include in a model more than 4 predictors). **Is there a similar recommendation concerning ANCOM-BC?**   

Moreover, broadly I am interested in analyzing the effect of physiological stress (covariate) on the differential abundance of gut microbiome in monkeys, correcting for the effect of age (categorical variable: infants, juveniles, adults), group of subjects (categorical: group1,2,3), and season of sample collection (categorical: summer, winter). Therefore I have a dataset of **2 samples per subject**, a winter sample and a summer sample and I was wondering if that's a problem considering that ANCOM-BC doesn't include random effects for longitudinal analysis.

Can I still do it?

Thank you again

Simone

**Comment by FrederickHuangLin** [comment, marked as answer]
Hi @simone-anza,

Thanks for your inquiry!

In general, ANCOM-BC is a linear model with an offset term serving for the bias correction purpose. Therefore, it is more flexible than some other complex models such as GLMM on the "limit" of the number of covariates. However, it is always highly recommended to justify the inclusion of covariates in the model. These covariates should be biologically relevant, and "over adjustment" will result in loss of power.

I understand your concern about repeated measures, and ANCOM-BC does not support mixed-effects models at this moment. However, given your sample size is limited (2 samples per subject), it won't fit well in a mixed-effects model. It is usually inadvisable to fit an effect as random if <5 degrees of freedom. I would suggest treating them as independent samples or computing the difference between seasons for each sample.

Best,
Huang
